src/attr/_make.py

```diff
@@ -1632,30 +1640,12 @@ def _generate_unique_filename(cls, func_name):
     """
     Create a "filename" suitable for a function being generated.
     """
-    unique_id = uuid.uuid4()
-    extra = ""
-    count = 1
-
-    while True:
-        unique_filename = "<attrs generated {0} {1}.{2}{3}>".format(
-            func_name,
-            cls.__module__,
-            getattr(cls, "__qualname__", cls.__name__),
-            extra,
-        )
-        # To handle concurrency we essentially "reserve" our spot in
-        # the linecache with a dummy line.  The caller can then
-        # set this value correctly.
-        cache_line = (1, None, (str(unique_id),), unique_filename)
-        if (
-            linecache.cache.setdefault(unique_filename, cache_line)
-            == cache_line
-        ):
-            return unique_filename
-
-        # Looks like this spot is taken. Try again.
-        count += 1
-        extra = "-{0}".format(count)
+    unique_filename = "<attrs generated {0} {1}.{2}>".format(
```

Comment:
I noticed that `_make_hash` does this:
```
    unique_filename = _generate_unique_filename(cls, "hash")
    type_hash = hash(unique_filename)
    ...
```
Does the fact that `type_hash` is no longer unique matter?  I don't think it does but I figure better to ask.

Reply:
Good question. Given two identical classes, equal instances will now hash the same. They will still not compare the same, I believe, so the contract is preserved. Not sure if this affects anything in the real world though. Maybe makes dicts and sets of identical classes a tiny bit less efficient?